website/addons/owncloud/views.py

```diff
@@ -0,0 +1,173 @@
+"""Views for the node settings page."""
+# -*- coding: utf-8 -*-
+import httplib as http
+
+from flask import request
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.addons.base import generic_views
+from owncloud import Client as OwnCloudClient
+from website.addons.owncloud.model import OwnCloudProvider
+from website.addons.owncloud.settings import DEFAULT_HOSTS
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.oauth.models import ExternalAccount
+from website.project.decorators import (
+    must_have_addon)
+from website.util import rubeus, api_url_for
+
+from website.addons.owncloud.utils import ExternalAccountConverter
+
+SHORT_NAME = 'owncloud'
+FULL_NAME = 'OwnCloud'
+
+owncloud_account_list = generic_views.account_list(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+owncloud_get_config = generic_views.get_config(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+def owncloud_root_folder(node_settings, auth, **kwargs):
```

Comment:
With a small model change, you should be able to use `generic_views.root_folder` for this